### PR TITLE
ICU-21957 fix NumberFormatterSettings::unitDisplayCase status, remove FormattedNumber:getGender

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-21957
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

Fix some status problems in numberformatter.h found during the BRS task to check non-stable API macros. Per TC discussion 2022-09-15:
- For NumberFormatterSettings::unitDisplayCase, restore the status to `@internal` with a note indicating future removal.
- For FormattedNumber:getGender, remove it now (along with implementation and test)

### CODE REVIEW

**Review by markusicu — APPROVED** [review]
lgtm tnx

File: icu4c/source/i18n/unicode/numberformatter.h [hunk 1]
```diff
@@ -2275,24 +2275,30 @@ class U_I18N_API NumberFormatterSettings {
     Derived displayOptions(const DisplayOptions &displayOptions) &&;
 #endif // U_HIDE_DRAFT_API
 
-#ifndef U_HIDE_DEPRECATED_API
+#ifndef U_HIDE_INTERNAL_API
     /**
-     * NOTE: This function is deprecated, use `displayOptions` instead.
+     * NOTE: Use `displayOptions` instead. This method was part of
+     * an internal technology preview in ICU 69, but will be removed
+     * in ICU 74, in favor of `displayOptions`
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/numberformatter.h:2282]:
```suggestion
     * in ICU 73, in favor of `displayOptions`
```

File: icu4c/source/i18n/unicode/numberformatter.h [hunk 2]
```diff
@@ -2275,24 +2275,30 @@ class U_I18N_API NumberFormatterSettings {
...
      *
-     * @deprecated ICU 72
+     * @internal
      */
     Derived unitDisplayCase(StringPiece unitDisplayCase) const &;
 
     /**
+     * NOTE: Use `displayOptions` instead. This method was part of
+     * an internal technology preview in ICU 69, but will be removed
+     * in ICU 74, in favor of `displayOptions`
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/numberformatter.h:2294]:
```suggestion
     * in ICU 73, in favor of `displayOptions`
```

**Review by markusicu — CHANGES_REQUESTED** [review]
actually, we said we would remove the tech preview apis in 73. they were never stable, so we are not replacing something stable with something draft.

File: icu4c/source/i18n/unicode/numberformatter.h (hunk 1 above)
Comment (pedberg-icu) [reply icu4c/source/i18n/unicode/numberformatter.h]: Thanks, changed

File: icu4c/source/i18n/unicode/numberformatter.h (hunk 2 above)
Comment (pedberg-icu) [reply icu4c/source/i18n/unicode/numberformatter.h]: Thanks, changed

**Review by markusicu — APPROVED** [review]
lgtm tnx again

**Comment by pedberg-icu** [comment]
Bizarre. The one test failure is for ICU4J, but this PR did not make any changes for ICU4J.

**Review by sffc — APPROVED** [review]

**Comment by pedberg-icu** [comment]
Re-running the tests fixed the failure.

**Pull request merged**